### PR TITLE
Feature/issue 07

### PR DESCRIPTION
**Pull request by kevin-verschaeve**

### CODE REVIEW

**Comment by Nicofuma** [comment]
You have a conflict with feature/issue-06 and indeed, we can see all the commits of the other branch. You should rebase to clean this stuff.

File: app/config/security.yml [hunk 1]
```diff
@@ -4,16 +4,10 @@ security:
...
+            connect_memory:
                 users:
-                    admin:
-                        password: admin
-                        roles: 'ROLE_ADMIN'
-        #sensiolabs_connect:
-            #connect_memory:
-                #users:
-                    #36221b12-f362-4dca-8724-4da82ef2ba49: "ROLE_ADMIN" # Sarah Khalil
+                    ebfc285e-727e-467e-b4f8-541057c05b6f: "ROLE_ADMIN"
```
Comment (jderusse) [inline app/config/security.yml]: Admin's UUID could be parameterized

File: composer.json [hunk 2]
```diff
@@ -25,12 +25,13 @@
         "symfony/assetic-bundle": "2.3.*",
         "symfony/monolog-bundle": "~2.4",
         "symfony/swiftmailer-bundle": "2.3.*",
-        "symfony/symfony": "~2.7",
+        "symfony/symfony": "2.7.*",
```
Comment (jderusse) [inline composer.json]: Why? you don't like 2.8 ?

File: src/SensioLabs/JobBoardBundle/Controller/BaseController.php [hunk 3]
```diff
@@ -25,21 +25,28 @@ class BaseController extends Controller
...
+        $country = $request->query->getAlpha('country');
+        $contractType = $request->query->getAlpha('contract_type');
         $page = $request->query->getInt('page', 1);
 
-        $jobs = $this->getDoctrine()
-                ->getManager()
-                ->getRepository('SensioLabsJobBoardBundle:Job')
-                ->findNextPosts($page);
+        $filter = $this->get('sensiolabs.manager.job');
+        $results = $filter->getFilters();
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/BaseController.php]: So, `getFilters` returns an array of `results`

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 4]
```diff
@@ -15,37 +16,74 @@
...
+            $this->getDoctrine()
+                ->getManager()
+                ->flush();
 
-        $job = $jobRepo->findOneBy(array(
-            'country' => $countryCode,
-            'contractType' => $contractType,
-            'slug' => $slug,
-        ));
+            $message = \Swift_Message::newInstance()
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: Belongs to the model

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 5]
```diff
@@ -15,37 +16,74 @@
...
+                ->setSubject('Job updated')
+                ->setFrom('noreply@sl-jobs.com')
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: should be parameterized

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 6]
```diff
@@ -15,37 +16,74 @@
...
+                ->setTo('kevin.verschaeve@sensiolabs.com')
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: should be parameterized

File: src/SensioLabs/JobBoardBundle/Entity/Job.php [hunk 7]
```diff
@@ -99,6 +99,21 @@ class Job
...
+     * @ORM\Column(type="boolean", options={"default" = 0})
+     */
+    private $isPublished;
+
+    /**
+     * Job constructor.
+     */
+    public function __construct()
+    {
+        $this->isPublished = 0;
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: not really a boolean

File: src/SensioLabs/JobBoardBundle/Entity/Job.php [hunk 8]
```diff
@@ -99,6 +99,21 @@ class Job
     private $slug;
 
     /**
+     * @var boolean
+     *
+     * @ORM\Column(type="boolean", options={"default" = 0})
+     */
+    private $isPublished;
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: name it `$published` and public methods `setPublished`, `isPublished`

File: src/SensioLabs/JobBoardBundle/Entity/JobRepository.php [hunk 9]
```diff
@@ -15,23 +15,65 @@ class JobRepository extends EntityRepository
...
     {
         $first = 0;
         if ($page > 1) {
             $first = ($page - 1) * self::NB_JOBS;
         }
 
-        return $this->createQueryBuilder('p')
-            ->setFirstResult($first)
+        $query = $this->createQueryBuilder('j')
+            ->where('j.isPublished = 1');
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Entity/JobRepository.php]: isPublished is a boolean not an integer

File: src/SensioLabs/JobBoardBundle/Entity/JobRepository.php [hunk 10]
```diff
@@ -15,23 +15,65 @@ class JobRepository extends EntityRepository
...
     }
+
+    /**
+     * @return array
+     */
+    public function findUsedCoutries()
+    {
+        return $this->createQueryBuilder('j')
+            ->select('j.country, COUNT(j.id) as nb')
+            ->where('j.isPublished = 1')
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Entity/JobRepository.php]: boolean

File: src/SensioLabs/JobBoardBundle/Entity/JobRepository.php [hunk 11]
```diff
@@ -15,23 +15,65 @@ class JobRepository extends EntityRepository
...
+    }
+
+    /**
+     * @return array
+     */
+    public function findUsedContractTypes()
+    {
+        return $this->createQueryBuilder('j')
+            ->select('j.contractType, COUNT(j.id) as nb')
+            ->where('j.isPublished = 1')
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Entity/JobRepository.php]: same here

File: src/SensioLabs/JobBoardBundle/Form/Type/JobType.php [hunk 12]
```diff
@@ -21,18 +21,16 @@ class JobType extends AbstractType
...
+            ->add('city', 'text', ['error_bubbling' => true])
+            ->add('contractType', 'choice', ['choices' => Job::$contractTypesChoices, 'error_bubbling' => true])
+            ->add('description', 'textarea', ['error_bubbling' => true])
+            ->add('howToApply', 'text', ['required' => false])
             ->add('Preview', 'submit', [
                 'label' => $builder->getData()->getId() ? 'Update' : 'Preview',
-                'attr' => [
-                    'class' => 'green-button right submit-btn'
-                ]
+                'attr' => ['class' => 'green-button right submit-btn'],
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Form/Type/JobType.php]: belongs to the view

File: src/SensioLabs/JobBoardBundle/Form/Type/JobType.php [hunk 13]
```diff
@@ -21,18 +21,16 @@ class JobType extends AbstractType
...
-            ->add('city', 'text', ['error_bubbling' => true,])
-            ->add('contractType', 'choice', ['choices' => Job::$contractTypesChoices, 'error_bubbling' => true,])
-            ->add('description', 'textarea', ['error_bubbling' => true,])
-            ->add('howToApply', 'text', ['required' => false,])
+            ->add('title', 'text', ['error_bubbling' => true])
+            ->add('company', 'text', ['error_bubbling' => true])
+            ->add('country', 'country', ['error_bubbling' => true])
+            ->add('city', 'text', ['error_bubbling' => true])
+            ->add('contractType', 'choice', ['choices' => Job::$contractTypesChoices, 'error_bubbling' => true])
+            ->add('description', 'textarea', ['error_bubbling' => true])
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Form/Type/JobType.php]: E_TOOMANY_BUBBLE

File: src/SensioLabs/JobBoardBundle/Resources/public/css/custom.css [hunk 14]
```diff
@@ -9,12 +9,9 @@
...
 }
 
-#error > ul > li {
-    color: #F00;
-}
-
+#error > ul > li ,
 .error_text {
-    color: #FOO;
+    color: #F00;
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/public/css/custom.css]: lol

**Pull request closed without merging**